db/schema.rb

```diff
@@ -466,6 +466,7 @@
     t.boolean "enable_job_applications"
     t.string "completed_steps", default: [], null: false, array: true
     t.string "actual_salary"
+    t.text "working_patterns_details"
```

Comment:
This is more of a general question than a critique of what you've decided here, but is there any reason that we should ever use `string` instead of `text`? There is some discussion [here](https://stackoverflow.com/questions/3354330/difference-between-string-and-text-in-rails). There is no performance penalty in postgres for using text instead of string, but we might want to change database one day.

Reply:
I (personally) prefer using both `string` and `text` to imply that one takes a bit of text, and the other a lot - but that's mostly because I came of age in ancient times when MySQL was all the rage and DB space was at a premium 👴 

I think using `text` across the board is probably fine.

Reply:
I went with text as I saw other text_area fields had their corresponding attributes set to this type. From looking at the post you linked to I can see string maps to the type VARCHAR(255) in MySQL, with a limit of 255 characters I assume? text maps to a type with a limit of 4294967296, as far as I can see from that post?

I read that string, when mapped to a datatype in postgres (character varying(n)) has a much bigger limit, is that correct?

If that's the case and I'm understanding things correctly, if we're concerned about problems arising from changing database, surely using text makes sense (if the same limit exists to the string's mapped types in other databases)?

Reply:
Please let me know if I haven't made any sense there 🤣